packages/@o3r/store-sync/src/sync-storage/storage-sync.ts

```diff
@@ -328,7 +332,7 @@ export const syncStorage = (config: SyncStorageConfig) => (reducer: any) => {
 
     // If state arrives undefined, we need to let it through the supplied reducer
     // in order to get a complete state as defined by user
-    if (action.type === INIT && !state) {
```

Comment:
Maybe we should check `syncForFeature` when `action.type` is not `INIT`

I don't understand why we need `RECOMPUTE`
because `UPDATE` will be called when loading the lazy loaded store, and it should be enough no ?
`RECOMPUTE` seems to be for devtools so in production mode it will not be called no ?

Reply:
Added the check. Also, for the second question, we need  `RECOMPUTE` because in dev mode, we get an undefined state for both `UPDATE` and `RECOMPUTE` action types even if the reducer is invoked for `UPDATE`, the state arrives as undefined for `RECOMPUTE`. Hence, the app breaks and is not usable.

Please check the reproduction: https://github.com/Abhirocks889/store-sync-error